Objects/obmalloc.c

```diff
@@ -1231,6 +1231,16 @@ _PyObject_XDecRefDelayed(PyObject *ptr)
 }
 #endif
 
+#ifdef Py_GIL_DISABLED
+void
+_PyObject_XSetRefDelayed(PyObject **ptr, PyObject *value)
+{
+    PyObject *old = *ptr;
+    FT_ATOMIC_STORE_PTR_RELEASE(*ptr, Py_NewRef(value));
```

Comment:
The default build implementation in `pycore_pymem.h` doesn't include the `Py_NewRef()`.

My preference is to keep the `Py_NewRef(value)` in the callers, i.e.:

```c
_PyObject_XSetRefDelayed(dictptr, Py_NewRef(value));
```

to match the semantics of `Py_XSETREF`.